### PR TITLE
Release 2.0 - change to an ansible collection

### PR DESCRIPTION
**Pull request by aalaesar**
Preparing release
@staticdev can the release drafter manage all the changes here or should we edit it ? I noted that is has overwritten my comments on the draft release last time.

### CODE REVIEW

**Comment by staticdev** [comment]
> Preparing release @staticdev can the release drafter manage all the changes here or should we edit it ? I noted that is has overwritten my comments on the draft release last time.

Release manager considers all PRs to main, except with tag `skip-changelog` as one change. In this case, makes sense to edit manually since it is a special case.

**Comment by aalaesar** [comment]
> > Preparing release @staticdev can the release drafter manage all the changes here or should we edit it ? I noted that is has overwritten my comments on the draft release last time.
> 
> Release Drafter considers all PRs to main, except with tag `skip-changelog` as one change. In this case, makes sense to edit manually since it is a special case.

Ok but I have a doubt : last time I edited the release draft to add a comment that this was the last release as a role, but this comment is not in the 1.8.0 release.
unless someone removed it ?

**Comment by aalaesar** [comment]
I expect troubles with galaxy when this PR is merged. 

I haven't found doc about converting a role into a collection for galaxy (and keeping stats)
could be a nice exp for when transferring to nextcloud (but it will have a larger exposure so good stats may not be that hard to get)

I didn't found doc about transferring a role in galaxy to an other organization either...

**Comment by staticdev** [comment]
> I expect troubles with galaxy when this PR is merged.
> 
> I haven't found doc about converting a role into a collection for galaxy (and keeping stats) could be a nice exp for when transferring to nextcloud (but it will have a larger exposure so good stats may not be that hard to get)
> 
> I didn't found doc about transferring a role in galaxy to an other organization either...

@aalaesar I also had trouble with that. Probably there was a merge of PR after your edit and then Release-Drafter overwritten it. This is why currently I edit after the release to be 100% safe =p

**Review by wiktor2200 — CHANGES_REQUESTED** [review]
I've taken a look on whole PR. I was able to verify all file moves, yml and other. I've not verified python plugins (I think nothing has changed since last review by @staticdev).

File: README.md
Comment (wiktor2200) [inline README.md:38]: Maybe it's worth adding link to role here:
```suggestion
[aalaesar.nextcloud.install_nextcloud](roles/install_nextcloud)|Install and configure an Nextcloud instance for a Debian/Ubuntu server - formerly `aalaesar.install_nextcloud`
```

File: galaxy.yml
Comment (aalaesar) [reply galaxy.yml:43]: Hmm.
I'm cautious with how ansible deal with dependencies declared in the meta.
As for roles, it implies that ansible run the roles declared as dependency first
I'll look at Galaxy doc first. And will do some test install and run.
Could be worth a PR.

Comment (wiktor2200) [reply galaxy.yml:43]: Sure, makes sense. It would be bad to run whole external role with default values accidentally, without any flow control :)

Comment (aalaesar) [reply galaxy.yml:43]: I've check ove the [galaxy doc](https://docs.ansible.com/ansible/devel/dev_guide/collections_galaxy_meta.html#collections-galaxy-meta)
looks like this field is only dedicated for collections, not roles
SO they may be an issue when installing the collection with the ansible-galaxy install command that will not pull the php_repository role from geerlinguy

Comment (aalaesar) [reply galaxy.yml:43]: found https://github.com/ansible/ansible/issues/76030#issuecomment-1137238958
it doesn't seems to be possible to depends on one stand alone role.
We nay need to use [jeff's php collection](https://github.com/geerlingguy/ansible-collection-php_roles) as dependency instead

**Comment by staticdev** [comment]
@aalaesar I think you will need somehow to rebase to be able to merge since Github says there is a conflict in meta/main.yml

File: galaxy.yml
Comment (wiktor2200) [reply galaxy.yml:43]: Hmm... I can see. It's a mess there in Ansible repos and documentation. I guess using whole collection as dependency could be dangerous (or at least decreasing readability).
Adding Jeff's submodule to our `roles` directory is not ideal, but may be one of the solutions.
The other one, could be skip defining `dependencies` now. Then see what will happen later when collection would be released and we will see if role's dependencies are resolved while trying to install with `galaxy install` command.

Comment (aalaesar) [reply galaxy.yml:43]: Hmm. I think that, at least, the requirement.yml and dependency  chapter from the role's readme should be moved to the collection root folder and readme.
So there would be a minimal amount of documentation to install completely the collection.

Comment (staticdev) [reply galaxy.yml:43]: Strange.. can't we have both link in here?
https://github.com/staticdev/linux-workstation-playbook/blob/main/requirements.yml

Comment (wiktor2200) [reply galaxy.yml:43]: I really like @staticdev's solution :)
I also agree with @aalaesar  that installation of collection should be doable with "minimal" effort with fail-safe default parameters and dependencies.

Comment (aalaesar) [reply galaxy.yml:43]: > Strange.. can't we have both link in here? https://github.com/staticdev/linux-workstation-playbook/blob/main/requirements.yml

yes, that the point.
that pretty much all what we can do to help customer install dependencies.
we can't use the "automated way" that uses deps in galaxy.yml

Comment (aalaesar) [reply galaxy.yml:43]: I've updated the readme to give instruction about installing standalone roles used by the collection

**Comment by staticdev** [comment]
Lets go.

File: galaxy.yml
Comment (aalaesar) [reply galaxy.yml:43]: @wiktor2200 one last review before the big merge ?

**Review by wiktor2200 — APPROVED** [review]
LGTM :)

**Pull request merged**
